sdk/core/src/http_client/noop.rs

```diff
@@ -3,6 +3,10 @@ use async_trait::async_trait;
 #[derive(Debug)]
 pub struct NoopClient;
 
+pub fn new_noop_client() -> std::sync::Arc<dyn crate::HttpClient> {
```

Comment:
I checked out this PR, as I wanted to investigate how these changes might play out with some of my other projects.  I noticed for the unit test which this function is used, using `new_http_client` (the primary mechanism to expose HTTP clients in the SDK) does not result in a change in the tests.

Why not use that, rather than have a new public API we have to account for the compatibility the future?

Reply:
That is a good point. Generally, how can one expose test functionality to another crate is the same workspace? It does validate that no network calls are made in the create tests. I am okay with backing this out.

Reply:
Unfortunately, the only way to expose functionality from one crate to another is with a public API.  

That said, in `azure_core`, we use the module name `azure_core::__private` to re-export the `paste` macro to the other crates as it's used by the `operation!` implementation which we export as a public API.  

https://github.com/Azure/azure-sdk-for-rust/blob/9cd67babbfccb22f7c4e6ab989123b6145869ba9/sdk/core/src/lib.rs#L110-L114

In my opinion, we should be careful about adding additional private-by-name-only APIs.  As Windows backwards compatibility testing has repeatedly shown, even if something is named "dont_use_this_as_its_badly_broken", people will depend on it.  (obligatory XKCD ref: https://xkcd.com/1172/)

Reply:
It has been backed out. Some API surfaces are more stable than others. I think naming the features and modules appropriately is good communication.